### PR TITLE
Bluetooth: Mesh: Fix unref null pointer

### PR DESCRIPTION
**Pull request by LingaoM**
When pb-gatt advertising enabled, after establish connection,
will call `cb->connected` and `cb->adv_send`.

In previous connected also clear `ADV_FLAG_PROXY` flag, but
in `adv_send` will attempt unref null point buffers.

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

### CODE REVIEW

File: subsys/bluetooth/mesh/adv_ext.c [hunk 1]
```diff
@@ -343,8 +343,9 @@ static void adv_sent(struct bt_le_ext_adv *instance,
 	BT_DBG("Advertising stopped after %u ms", (uint32_t)duration);
 
 	atomic_clear_bit(adv->flags, ADV_FLAG_ACTIVE);
+	atomic_clear_bit(adv->flags, ADV_FLAG_PROXY);
 
-	if (!atomic_test_and_clear_bit(adv->flags, ADV_FLAG_PROXY)) {
+	if (adv->buf) {
 		net_buf_unref(adv->buf);
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/adv_ext.c:349]: Perhaps we could clear adv->buf here now too, to ensure there are no regressions for this check in the future?

Comment (LingaoM) [reply subsys/bluetooth/mesh/adv_ext.c:349]: Logically, calling here is already an error

**Review by trond-snekvik — DISMISSED** [review]

File: subsys/bluetooth/mesh/adv_ext.c [hunk 2]
```diff
@@ -208,6 +210,29 @@ static void send_pending_adv(struct k_work *work)
 	int err;
 
 	adv = CONTAINER_OF(work, struct bt_mesh_ext_adv, work.work);
+
+	if (atomic_test_and_clear_bit(adv->flags, ADV_FLAG_SENDED)) {
```
Comment (LingaoM) [inline subsys/bluetooth/mesh/adv_ext.c:214]: @trond-snekvik Please note that this is a new bug fix that I put in the same PR as it is associated with the first commit.

File: subsys/bluetooth/mesh/adv_ext.c [hunk 3]
```diff
@@ -33,6 +33,8 @@
 enum {
 	/** Controller is currently advertising */
 	ADV_FLAG_ACTIVE,
+	/** Advertising sending completed */
+	ADV_FLAG_SENDED,
```
Comment (alxelax) [inline subsys/bluetooth/mesh/adv_ext.c:37]:
```suggestion
	ADV_FLAG_SENT,
```

**Review by alxelax — APPROVED** [review]

**Review by trond-snekvik — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**